### PR TITLE
Implemented soft delete concept

### PR DESCRIPTION
**Pull request by Lessorg**

### CODE REVIEW

File: src/main/java/test/project/bookshop/model/Book.java [hunk 1]
```diff
@@ -9,10 +9,14 @@
...
 import lombok.Getter;
 import lombok.Setter;
+import org.hibernate.annotations.SQLDelete;
+import org.hibernate.annotations.Where;
 
 @Setter
 @Getter
 @Entity
+@SQLDelete(sql = "UPDATE books SET is_deleted = true WHERE id=?")
+@Where(clause = "is_deleted=false")
```
Comment (Elena-Bruyako) [inline src/main/java/test/project/bookshop/model/Book.java]: check Hibernate version if this annotation deprecated or not

File: src/main/java/test/project/bookshop/service/impl/BookServiceImpl.java [hunk 2]
```diff
@@ -31,8 +31,31 @@ public List<BookDto> findAll() {
...
+    public BookDto delete(Long id) {
+        Book existingBook = bookRepository.findById(id).orElseThrow(
+                () -> new EntityNotFoundException("Can't find book by id: " + id));
+        bookRepository.deleteById(id);
+        return bookMapper.toDto(existingBook);
+    }
+
+    @Override
+    public BookDto update(Long id, BookRequestDto requestDto) {
+        Book existingBook = bookRepository.findById(id).orElseThrow(
```
Comment (Elena-Bruyako) [inline src/main/java/test/project/bookshop/service/impl/BookServiceImpl.java]:
```suggestion
        Book book = bookRepository.findById(id).orElseThrow(
```

File: src/main/java/test/project/bookshop/service/impl/BookServiceImpl.java [hunk 3]
```diff
@@ -31,8 +31,31 @@ public List<BookDto> findAll() {
...
+                () -> new EntityNotFoundException("Can't find book by id: " + id));
+
+        existingBook.setTitle(requestDto.getTitle());
```
Comment (Elena-Bruyako) [inline src/main/java/test/project/bookshop/service/impl/BookServiceImpl.java]: use Mapper to update book
`bookMapper.updateBookFromDto(bookDto, book);`

File: src/main/resources/db/changelog/changes/01-create-books-table.yaml [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+              - column:
+                  name: cover_image
+                  type: text
+              - column:
+                  name: is_deleted
+                  type: bool
+                  constraints:
+                    nullable: false
+        - sql:
+            sql: "ALTER TABLE books ALTER COLUMN is_deleted SET DEFAULT false"
```
Comment (Elena-Bruyako) [inline src/main/resources/db/changelog/changes/01-create-books-table.yaml]: why do you need this sql? remove it

File: src/main/java/test/project/bookshop/service/impl/BookServiceImpl.java [hunk 5]
```diff
@@ -31,8 +31,31 @@ public List<BookDto> findAll() {
...
-    public BookDto findByid(Long id) {
-        return bookMapper.toDto(bookRepository.findByid(id).orElseThrow(
+    public BookDto findById(Long id) {
+        return bookMapper.toDto(bookRepository.findById(id).orElseThrow(
                 () -> new EntityNotFoundException("Can't find book by id: " + id)));
     }
+
+    @Override
+    public BookDto delete(Long id) {
+        Book existingBook = bookRepository.findById(id).orElseThrow(
```
Comment (Elena-Bruyako) [inline src/main/java/test/project/bookshop/service/impl/BookServiceImpl.java]: you can check existing book by creating method in Repo

**Review by Elena-Bruyako — CHANGES_REQUESTED** [review]
see comments

File: src/main/resources/db/changelog/changes/01-create-books-table.yaml (hunk 4 above)
Comment (Lessorg) [reply src/main/resources/db/changelog/changes/01-create-books-table.yaml]: I haven't found another way to set the "is-deleted" default column value to false

File: src/main/java/test/project/bookshop/service/impl/BookServiceImpl.java [hunk 6]
```diff
@@ -31,8 +31,31 @@ public List<BookDto> findAll() {
...
+    public BookDto findById(Long id) {
+        return bookMapper.toDto(findBookById(id));
+    }
+
+    @Override
+    public BookDto delete(Long id) {
+        if (!bookRepository.existsById(id)) {
+            throw new EntityNotFoundException("Book with id " + id + " does not exist.");
+        }
+        Book book = findBookById(id);
```
Comment (Elena-Bruyako) [inline src/main/java/test/project/bookshop/service/impl/BookServiceImpl.java]:
```suggestion
```

File: src/main/java/test/project/bookshop/service/impl/BookServiceImpl.java [hunk 7]
```diff
@@ -31,8 +31,31 @@ public List<BookDto> findAll() {
...
+
+        bookRepository.deleteById(id);
+        return bookMapper.toDto(book);
+    }
+
+    @Override
+    public BookDto update(Long id, BookRequestDto requestDto) {
+        Book book = findBookById(id);
+
+        book = bookMapper.updateBookFromDto(requestDto, book);
```
Comment (Elena-Bruyako) [inline src/main/java/test/project/bookshop/service/impl/BookServiceImpl.java]:
```suggestion
        bookMapper.updateBookFromDto(requestDto, book);
```

File: src/main/java/test/project/bookshop/mapper/BookMapper.java [hunk 8]
```diff
@@ -10,5 +12,11 @@
 public interface BookMapper {
     BookDto toDto(Book book);
 
+    @Mapping(target = "id", ignore = true)
+    @Mapping(target = "deleted", ignore = true)
     Book toBook(BookRequestDto bookRequestDto);
+
+    @Mapping(target = "id", ignore = true)
+    @Mapping(target = "deleted", ignore = true)
+    Book updateBookFromDto(BookRequestDto bookDto, @MappingTarget Book book);
```
Comment (Elena-Bruyako) [inline src/main/java/test/project/bookshop/mapper/BookMapper.java]:
```suggestion
    void updateBookFromDto(BookRequestDto bookDto, @MappingTarget Book book);
```

**Review by Elena-Bruyako — CHANGES_REQUESTED** [review]
see comments

**Review by Elena-Bruyako — APPROVED** [review]

**Pull request merged**